src/widgets.js

```diff
@@ -591,11 +728,51 @@ define([
         return $el;
     }
 
+    function replaceOuputRef(form, value, withClose, noOutput) {
+        var template = "<output value=\"" + value + "\" />";
+        if (noOutput) {
+            template = value;
+        }
+        var v = value.split('/'),
+            dispValue = v[v.length-1],
+            mug = form.getMugByPath(value),
+            icon = mug ? mug.options.icon: 'fcc fcc-fd-external-case',
+            datanodeClass = mug ? 'label-datanode-internal' : 'label-datanode-external',
+            richText = $('<span>').addClass('label label-datanode')
+                .addClass(datanodeClass)
+                .attr({
+                    contenteditable: false,
+                    draggable: true,
+                    'data-value': template
+                }).append($('<i>').addClass(icon).html('&nbsp;')).append(dispValue);
+        if (withClose) {
+            richText.append($("<button>").addClass('close').html("&times;"));
+        }
+        return richText;
+    }
+
+    function toRichText(val, form, withClose) {
+        if (!val) {return "";}
+        val = val.replace('&lt;', '<').replace('&gt;', '>');
```

Comment:
Yeah. Something more obvious should be used here. All this is currently doing is replacing this data-value https://github.com/dimagi/Vellum/pull/493/files#diff-e5b3eca8407b45b132e0147b8392192cR845 and I wanted it to be specific so that I ran into less weirdness here. I plan on using something better in the future once the formats are more fleshed out.